pkg/definition/gen_sdk/gen_sdk.go

```diff
@@ -0,0 +1,494 @@
+	if err := g.def.FromCUEString(string(cueBytes), g.meta.config); err != nil {
+		return nil, errors.Wrapf(err, "failed to parse CUE")
+	}
+	g.name = g.def.GetName()
+	g.kind = g.def.GetKind()
+
+	templateString, _, err := unstructured.NestedString(g.def.Object, definition.DefinitionTemplateKeys...)
+	if err != nil {
+		return nil, err
+	}
+	template, err := value.NewValue(templateString+velacue.BaseTemplate, nil, "")
+	if err != nil {
+		return nil, err
+	}
+	return template, nil
+}
+
+// GenOpenAPISchema generates OpenAPI json schema from cue.Instance
+func (g *Generator) GenOpenAPISchema(val *value.Value) error {
+	var err error
+	defer func() {
+		if r := recover(); r != nil {
+			err = fmt.Errorf("invalid cue definition to generate open api: %v", r)
+			debug.PrintStack()
+			return
+		}
+	}()
+	if val.CueValue().Err() != nil {
+		return val.CueValue().Err()
+	}
+	paramOnlyVal, err := common.RefineParameterValue(val)
+	if err != nil {
+		return err
+	}
+	defaultConfig := &openapi.Config{ExpandReferences: false, NameFunc: func(val cue.Value, path cue.Path) string {
+		sels := path.Selectors()
+		lastLabel := sels[len(sels)-1].String()
+		return strings.TrimPrefix(lastLabel, "#")
+	}, DescriptionFunc: func(v cue.Value) string {
+		for _, d := range v.Doc() {
+			if strings.HasPrefix(d.Text(), "+usage=") {
+				return strings.TrimPrefix(d.Text(), "+usage=")
+			}
+		}
+		return ""
+	}}
+	b, err := openapi.Gen(paramOnlyVal, defaultConfig)
+	if err != nil {
+		return err
+	}
+	doc, err := openapi3.NewLoader().LoadFromData(b)
+	if err != nil {
+		return err
+	}
+
+	g.completeOpenAPISchema(doc)
+	openapiSchema, err := doc.MarshalJSON()
+	g.openapiSchema = openapiSchema
+	return err
+}
+
+func (g *Generator) completeOpenAPISchema(doc *openapi3.T) {
+	for key, schema := range doc.Components.Schemas {
+		switch key {
+		case "parameter":
+			spec := g.name + "-spec"
+			schema.Value.Title = spec
+			completeFreeFormSchema(schema)
+			completeSchemas(schema.Value.Properties)
+			doc.Components.Schemas[spec] = schema
+			delete(doc.Components.Schemas, key)
+		default:
+			completeSchema(key, schema)
+		}
+	}
+}
+
+// GenerateCode will call openapi-generator to generate code and modify it
+func (g *Generator) GenerateCode() (err error) {
+	tmpFile, err := os.CreateTemp("", g.name+"-*.json")
+	_, err = tmpFile.Write(g.openapiSchema)
+	if err != nil {
+		return errors.Wrap(err, "write openapi schema to temporary file")
+	}
+	defer func() {
+		_ = tmpFile.Close()
+		if err == nil {
+			_ = os.Remove(tmpFile.Name())
+		}
+	}()
+	apiDir, err := filepath.Abs(path.Join(g.meta.Output, "pkg", "apis"))
+	if err != nil {
+		return errors.Wrapf(err, "get absolute path of %s", apiDir)
+	}
+
+	// nolint:gosec
+	cmd := exec.Command("docker", "run",
+		"-v", fmt.Sprintf("%s:/local/output", apiDir),
+		"-v", fmt.Sprintf("%s:/local/input", filepath.Dir(tmpFile.Name())),
+		"-v", fmt.Sprintf("%s:/local/template", g.meta.templatePath),
```

Comment:
```suggestion
		"-v", fmt.Sprintf("%s:/local/template", g.meta.templatePath),
		"--rm",
```
maybe delete the container after operation to avoid leaving unused containers behind

Reply:
Here we are using the image multiple times, we'd better keep it here otherwise it will be pulled for times.

Reply:
`--rm` here will not delete the image, only the generated container from the image. So it will be as fast as before as long as the user don;t delete the image manually

Reply:
OK, you are right. `--rm` won't remove the image. It removes the container.